src/plugins/dashboard/public/services/dashboard_backup/dashboard_backup_service.ts

```diff
@@ -52,6 +55,21 @@ class DashboardSessionStorageService implements DashboardSessionStorageServiceTy
     }
   }
 
+  public getViewMode = (): ViewMode => {
+    return this.localStorage.get(DASHBOARD_VIEWMODE_LOCAL_KEY);
+  };
+
+  public storeViewMode = (viewMode: ViewMode) => {
+    try {
+      this.localStorage.set(DASHBOARD_VIEWMODE_LOCAL_KEY, viewMode);
+    } catch (e) {
+      this.notifications.toasts.addDanger({
+        title: panelStorageErrorStrings.getPanelsGetError(e.message),
+        'data-test-subj': 'dashboardPanelsGetFailure',
+      });
+    }
+  };
+
   public clearState(id = DASHBOARD_PANELS_UNSAVED_ID) {
     try {
       const sessionStorage = this.sessionStorage.get(DASHBOARD_PANELS_SESSION_KEY);
```

Comment:
Now that we are storing `viewMode` in local storage, do we still need it in session storage for each dashboard?

![image](https://github.com/elastic/kibana/assets/8698078/275f80b9-63e4-40a5-9ad8-8ca9a934c6cc)

Reply:
Unfortunately we do. The view mode for each Dashboard is also used to determine which ones show up in the listing page as having `unsaved changes`.

Reply:
Booo, that's too bad. Makes sense, though - didn't think about the listing page 🤦